outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+                today.getDayOfMonth(),
+                today.getHourOfDay(),
+                today.getMinuteOfHour(),
+                today.getSecondOfMinute());
+    }
+
+
+    private File createTargetFileDirectory() {
+        File userHome = new File(System.getProperty("user.home"));
+        File targetFileDirectory = new File(userHome, settingsFacade.getProperty(TARGET_FILE_DIRECTORY));
+
+        if (targetFileDirectory.exists()) {
+            LOGGER.info("targetFile directory exists: {}", targetFileDirectory);
+        } else {
+            LOGGER.info("creating targetFile directory: {}", targetFileDirectory);
+            if (!targetFileDirectory.mkdirs()) {
+                String error = String.format("Unable to create targetFileDirectory %s: mkdirs() failed",
+                        targetFileDirectory);
+                LOGGER.error(error);
+                alertService.create(targetFileDirectory.toString(), "targetFileDirectory", "mkdirs() failed",
+                        AlertType.CRITICAL, AlertStatus.NEW, 0, null);
+                throw new IllegalStateException();
+            }
+        }
+        return targetFileDirectory;
+    }
+
+
+    public TargetFileNotification generateTargetFile() {
+        String targetFileName = targetFileName();
+        File targetFileDirectory;
+        MessageDigest md;
+        int recordCount = 0;
+
+        try {
+            targetFileDirectory = createTargetFileDirectory();
+        } catch (IllegalStateException e) {
+            return null;
+        }
+
+        File targetFile = new File(targetFileDirectory, targetFileName);
+        try (FileOutputStream fos = new FileOutputStream(targetFile);
+             OutputStreamWriter writer = new OutputStreamWriter(fos)) {
+
+            md = MessageDigest.getInstance("MD5");
+            @SuppressWarnings("PMD.UnusedLocalVariable")
+            DigestOutputStream dos = new DigestOutputStream(fos, md);
+
+            //figure out which day to work with
+            final DayOfTheWeek today = DayOfTheWeek.today();
+
+
+            int maxQueryBlock = Integer.parseInt(settingsFacade.getProperty(MAX_QUERY_BLOCK));
+
+            //Fresh calls
+            int page = 1;
+            int numBlockRecord;
+            do {
+                //todo: replace retrieveAll with findByStatus when available
+                List<Subscription> subscriptions = subscriptionDataService.retrieveAll(
+                        new QueryParams(page, maxQueryBlock));
+                numBlockRecord = subscriptions.size();
+
+                for (Subscription subscription : subscriptions) {
+                    writer.write(subscription.getSubscriptionId());
+                    writer.write(",");
+                    writer.write(subscription.getSubscriber().getCallingNumber().toString());
+                    writer.write(",");
+                    writer.write(subscription.getLanguage().getCode());
+                    //todo...
```

Comment:
Yep, will do next checkin when I merge & have access to that property on the Subscription object.